web/pages/[[...slug]].tsx

```diff
@@ -21,7 +21,13 @@ const QUERY = groq`
       ...,
       page-> {
         name,
-        hero,
+        hero {
+          ...,
+          callToAction {
+            ...,
+            reference->{slug},
```

Comment:
```suggestion
            reference->,
```

Nit: I've tried to refrain from nesting the queries more than necessary. This suggestion includes more than just the `slug` field in the response now, if present, whereas the previous alternative would only include `slug`.

Reply:
Oh OK, I thought this was just done due to lack of time.

Only including `slug` is exactly what I was going for, because that's all we need (as per the `SimpleCallToAction` TS type). As far as I know, all extra/unused fields we query for here will add to the size of the `pageProps` JSON that is loaded client-side when navigating between pages.

https://www.sanity.io/blog/3-simple-things-in-groq-to-supercharge-your-frontend-development#db26fcd247c4

If you nevertheless prefer to leave the projection out (e.g. for consistency), I'll commit your suggestion, though. Just checking :)

Reply:
Yeah it's a bit less convenient for the phase where everything is very unstable, as that blog article says… :)